core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -220,11 +222,69 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
       writer.literal("INT64");
       writer.endFunCall(castFrame);
       break;
+    case REGEXP_SUBSTR:
+      unparseRegexSubstr(writer, call, leftPrec, rightPrec);
+      break;
     default:
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
 
+  private void unparseRegexSubstr(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    final SqlWriter.Frame regexFrame;
+    switch (call.operandCount()) {
+    case 2:
+      regexFrame = writer.startFunCall("REGEXP_EXTRACT");
+      for (SqlNode operand : call.getOperandList()) {
+        writer.sep(",");
```

Comment:
Can we do it in a better way , like 
  SqlSyntax.FUNCTION.unparse(writer, REGEXP_EXTRACT, call, leftPrec, rightPrec);

Reply:
I am unaware of this class. Will have a look at it.